### PR TITLE
Clang cevian

### PR DESCRIPTION
**Pull request by cevian**

### CODE REVIEW

File: src/agg_bookend.c [hunk 1]
```diff
@@ -34,24 +34,23 @@ TS_FUNCTION_INFO_V1(ts_bookend_deserializefunc);
 /* A  PolyDatum represents a polymorphic datum */
 typedef struct PolyDatum
 {
-	Oid			type_oid;
-	bool		is_null;
-	Datum		datum;
+	Oid type_oid;
```
Comment (erimatnor) [inline src/agg_bookend.c:36]: This still looks different. Spaces vs tabs?

File: src/agg_bookend.c [hunk 2]
```diff
@@ -395,12 +391,12 @@ ts_first_sfunc(PG_FUNCTION_ARGS)
 }
 
 /* last(internal internal_state, anyelement value, "any" comparison_element) */
-Datum
-ts_last_sfunc(PG_FUNCTION_ARGS)
+Datum ts_last_sfunc(PG_FUNCTION_ARGS)
```
Comment (erimatnor) [inline src/agg_bookend.c]: The `Datum` should be on its own line here.

**Review by erimatnor — COMMENTED** [review]
This looks like it is still changing the formatting, especially it looks like it is not doing tab spacing correctly. It also doesn't properly put a function's return type on its own line.

File: src/agg_bookend.c (hunk 1 above)
Comment (cevian) [reply src/agg_bookend.c:36]: The difference here is that fields are not aligned instead one space is entered after a field. Clang does not support aligned declarations with the point right format. So we'd have to have unaligned fields. I think this is fine but if you disagree we can't use clang

Comment (cevian) [reply src/agg_bookend.c:36]: The difference here is that fields are not aligned instead one space is entered after a field. Clang does not support aligned declarations with the point right format. So we'd have to have unaligned fields. I think this is fine but if you disagree we can't use clang

File: src/agg_bookend.c (hunk 2 above)
Comment (cevian) [reply src/agg_bookend.c]: I don't think clang supports this

Comment (cevian) [reply src/agg_bookend.c]: I don't think clang supports this

File: .clang-format [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    Priority:        -1
+IncludeIsMainRegex: '' # filename_<suffix> will be seen as the primary include
+IndentCaseLabels: true
+IndentPPDirectives: None # do not indent preprocessor directives after the '#'
+IndentWidth:     4
+IndentWrappedFunctionNames: false # we do not indent the function name in the declaration
+JavaScriptQuotes: Double # N/A js
+JavaScriptWrapImports: true # N/A js
+KeepEmptyLinesAtTheStartOfBlocks: false
+MacroBlockBegin: '' # regex of macros that behave like '{'
```
Comment (erimatnor) [inline .clang-format:81]: Shouldn't we put things like `PG_TRY()` here? Or does this not apply because it has braces?

File: src/bgw/job.c [hunk 4]
```diff
@@ -248,25 +263,23 @@ bgw_job_tuple_delete(TupleInfo *ti, void *data)
...
-		.tuplock = {
-			.waitpolicy = LockWaitBlock,
-			.lockmode = LockTupleExclusive,
-			.enabled = false,
-		}
-	};
+	Catalog *catalog = catalog_get();
+
+	ScannerCtx scanctx = { .table = catalog->tables[BGW_JOB].id,
+						   .index = CATALOG_INDEX(catalog, BGW_JOB, BGW_JOB_PKEY_IDX),
```
Comment (erimatnor) [inline src/bgw/job.c]: This looks pretty terrible now with Clang. Can we do something to fix it?

File: src/bgw/job.c [hunk 5]
```diff
@@ -21,12 +21,11 @@
...
 #include "telemetry/telemetry.h"
 
-#define TELEMETRY_INITIAL_NUM_RUNS	12
+#define TELEMETRY_INITIAL_NUM_RUNS 12
 
-const char *job_type_names[_MAX_JOB_TYPE] = {
-	[JOB_TYPE_VERSION_CHECK] = "telemetry_and_version_check_if_enabled",
-	[JOB_TYPE_UNKNOWN] = "unknown"
-};
+const char *job_type_names[_MAX_JOB_TYPE] = { [JOB_TYPE_VERSION_CHECK] =
```
Comment (erimatnor) [inline src/bgw/job.c]: Here it doesn't line-break the first member, which leads to weird formatting.

File: scripts/clang_format_wrapper.sh [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
```
Comment (erimatnor) [inline scripts/clang_format_wrapper.sh:2]: Add comment/documentation of why we have this script and what it is doing

File: .clang-format (hunk 3 above)
Comment (cevian) [reply .clang-format:81]: Does not apply because of braces. bgw/job.c has a block and it looks fine.

File: src/bgw/job.c (hunk 4 above)
Comment (cevian) [reply src/bgw/job.c]: Fixed. Designated initializers need a final comma at the last element, then clang-format fixes this.

File: src/bgw/job.c (hunk 5 above)
Comment (cevian) [reply src/bgw/job.c]: fixed with final comma

File: scripts/clang_format_wrapper.sh (hunk 6 above)
Comment (cevian) [reply scripts/clang_format_wrapper.sh:2]: fixed

**Pull request closed without merging**